product_attribute_model_link/views/product_attribute_views.xml

```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="product_attribute_model_link_form" model="ir.ui.view">
+        <field name="name">product.attribute.model.link.form</field>
+        <field name="model">product.attribute</field>
+        <field name="inherit_id" ref="product.product_attribute_view_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//group[@name='main_fields']" position="after">
+                <group>
+                    <group name="model_link_fields">
+                        <field name="linked_model_id" />
+                        <field
+                            name="linked_field_id"
+                            attrs="{'required': [('linked_model_id', '!=', False)]}"
+                        />
+                        <field name="domain" />
```

Comment:
Domain widget must be used here. Not a plain char field.
![image](https://github.com/OCA/product-attribute/assets/2234999/3dac38a7-ad77-4d1d-9131-346766668298)